Heber/src/main/java/springApp/java/com/example/demo/repositories/InstrumentoRepository.java

```diff
@@ -0,0 +1,63 @@
+package springApp.java.com.example.demo.repositories;
+
+import org.springframework.stereotype.Repository;
+import springApp.java.com.example.demo.exceptions.InstrumentoNoEncontradoException;
+import springApp.java.com.example.demo.models.AccionModel;
+import springApp.java.com.example.demo.models.BonoModel;
+import springApp.java.com.example.demo.models.InstrumentoFinancieroModel;
+
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public class InstrumentoRepository {
+
+    private List<InstrumentoFinancieroModel> instrumentosFinancieros = new ArrayList<>();
+    private Long currentId = 1L; // Variable para gestionar el ID autoincremental
+
+    public String helloRepository() {
+        return "Hello from Repository";
+    }
+
+    public AccionModel agregarAccion(AccionModel accion) {
+        accion.setId(currentId++); // Asignar el ID y luego incrementarlo
+        instrumentosFinancieros.add(accion);
+        return accion;
+    }
```

Comment:
En este punto un poco lo que vimos, no validamos que el objeto que llega no es null pero tambien tenemos la responsabilidad de incrementar el ID muy directamente, podriamos extraer el esto a metodos para quedarnos tranquilos de que tenemos centralizada el proceso y si algun cambio sucede lo tenemos unificado.

```suggestion
    public AccionModel agregarAccion(AccionModel accion) {
        if !ValidarAccion(accion) {
         //lanzar exception
         }
        IncrementarIdAccion();
        instrumentosFinancieros.add(accion);
        return accion;
    }
```